File: demo/src/main/java/com/example/demo/project/spring/entities/IssuingAuthority.java
package com.example.demo.project.spring.entities;


import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "ISSUING_AUTHORITY")
public class IssuingAuthority {

    @Id
    @GeneratedValue(generator = "gen_auth")
    @SequenceGenerator(name = "gen_auth", sequenceName = "auth_seq")
    private Integer id;

    @Column(name = "name", nullable = false)
    private String name;

    @JsonManagedReference
    @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL, mappedBy = "issuingAuthority")
    private List<City> city;

    @JsonBackReference
    @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL, mappedBy = "issuingAuthority")
    private List<IdCard> idCards;

    public IssuingAuthority() {}

    public IssuingAuthority(String name) {
        this.name = name;
    }

    public IssuingAuthority(String name, List<City> city) {
        this.name = name;
        this.city = city;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<City> getCity() {
        return city;
    }

    public void setCity(List<City> city) {
        this.city = city;
    }
}

File: angular/src/app/model/issuing-authority.ts
import {City} from "./city";
import {IdCard} from "./id-card";

export class IssuingAuthority {
  private _id:number
  private _name:String
  private _city:City[]
  private _idCards:IdCard[]

  constructor (obj?: any) {
    this._id = (obj && obj._id);
    this._name = (obj && obj._name);
    this._city = (obj && obj._city);
    this._idCards = (obj && obj._idCards);
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get name(): String {
    return this._name;
  }

  set name(value: String) {
    this._name = value;
  }

  get city(): City[] {
    return this._city;
  }

  set city(value: City[]) {
    this._city = value;
  }

  get idCards(): IdCard[] {
    return this._idCards;
  }

  set idCards(value: IdCard[]) {
    this._idCards = value;
  }
}
